bbq/compiler/compiler.go

```diff
@@ -3273,8 +3336,14 @@ func (c *Compiler[_, _]) VisitPragmaDeclaration(_ *ast.PragmaDeclaration) (_ str
 func (c *Compiler[_, _]) VisitImportDeclaration(declaration *ast.ImportDeclaration) (_ struct{}) {
 
 	var identifiers []ast.Identifier
+	// generate a map for aliases because when creating globals we need to associate each name with its alias
+	// each alias can be used multiple times, for example: for functions of the aliased type
+	aliases := make(map[string]string)
 	for _, imp := range declaration.Imports {
 		identifiers = append(identifiers, imp.Identifier)
+		if imp.Alias.Identifier != "" {
+			aliases[imp.Identifier.Identifier] = imp.Alias.Identifier
+		}
 	}
```

Comment:
With #4199 the alias map is now available from the elaboration so we can avoid re-allocating here, I'll push up a change

Reply:
Done in [1857524](https://github.com/onflow/cadence/pull/4125/commits/18575245d49ea5b597515fe76662fb3cc0c58d4b)